File: SpringBoot/SpringBoot-Java2/src/main/java/hello/core/MemberApp.java
package hello.core;

import hello.core.member.Grade;
import hello.core.member.Member;
import hello.core.member.MemberService;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;

public class MemberApp {
    public static void main(String[] args) {
        // AppConfig가 모든 의존성 주입을 담당.
//        AppConfig appconfig = new AppConfig();
//        MemberService memberService = appconfig.memberService();

        // Spring Framework를 사용한 의존성 주입
        ApplicationContext ctx = new AnnotationConfigApplicationContext(AppConfig.class);
        MemberService memberService = ctx.getBean("memberService", MemberService.class);

        Member memberA = new Member(1L, "memberA", Grade.VIP);
        memberService.join(memberA);

        Member foundMember = memberService.findMember(1L);
        System.out.println("new Member = " + memberA.getName());
        System.out.println("find Member = " + foundMember.getName());
    }
}

File: SpringBoot/SpringBoot-Kotlin/src/main/kotlin/hello/core/MemberApp.kt
package hello.core

import hello.core.member.Grade
import hello.core.member.Member
import hello.core.member.MemberRepository
import hello.core.member.MemberServiceImpl

class MemberApp

//fun main(args: Array<String>) {
//    val memberService = MemberServiceImpl(MemberRepository)
//    val member = Member(1L, "memberA", Grade.VIP)
//    memberService.join(member)
//
//    val findMember = memberService.findMember(1L)
//
//    // Test
//    println("new member = $member")
//    println("find member = $findMember")
//}